crates/world-id-core/src/authenticator.rs

```diff
@@ -237,3 +302,49 @@ impl Authenticator {
         unimplemented!()
     }
 }
+
+/// The inner signer which can sign requests for both on-chain and off-chain operations on behalf of the authenticator.
+///
+/// Both keys are zeroized on drop.
+#[derive(Debug)]
+pub struct AuthenticatorSigner {
+    /// An on-chain SECP256K1 private key.
+    onchain_signer: PrivateKeySigner,
+    /// An off-chain `EdDSA` private key.
+    offchain_signer: EdDSAPrivateKey,
+}
+
+impl AuthenticatorSigner {
+    /// Initializes a new signer from an input seed.
+    pub fn from_seed_bytes(seed: &[u8]) -> Result<Self> {
+        if seed.len() != 32 {
+            return Err(eyre::eyre!("seed must be 32 bytes"));
+        }
+        let bytes: [u8; 32] = seed.try_into()?;
+        let onchain_signer = PrivateKeySigner::from_bytes(&bytes.into())?;
+        let offchain_signer = EdDSAPrivateKey::from_bytes(bytes);
```

Comment:
is there any risk in reusing a seed for two different curves?

Reply:
i guess one seed compromise compromises both, but then we have to store multiple seeds